worker/raft/raftforwarder/metrics.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package raftforwarder
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricsNamespace = "juju_raftforwarder"
+)
+
+// metricsCollector is a prometheus.Collector that collects metrics
+// about how long it's taking to forward requests to raft and get
+// responses.
+type metricsCollector struct {
+	requests *prometheus.SummaryVec
+}
+
+func newMetricsCollector() *metricsCollector {
+	return &metricsCollector{
+		requests: prometheus.NewSummaryVec(prometheus.SummaryOpts{
+			Namespace: metricsNamespace,
+			Name:      "request",
+			Help:      "Request times for raft forwarder operations in ms",
+			Objectives: map[float64]float64{
+				0.5:  0.05,
+				0.9:  0.01,
+				0.99: 0.001,
+			},
+		}, []string{
+			// section can be "apply" (just the time to apply req and
+			// get a response from raft) or "full" (total time from
+			// request received to response sent)
+			"section",
+		}),
+	}
+}
+
+func (m *metricsCollector) record(start time.Time, section string) {
+	elapsedMS := float64(time.Now().Sub(start)) / float64(time.Millisecond)
```

Comment:
Why not just use `time.Now().Sub(start).Milliseconds()`, it returns an int64

Reply:
Not sure - I cargo-culted this from the raftlease store metrics - just looking to see if there's a reason.

Reply:
Duration.Milliseconds() is Go 1.13. I think we're still Go 1.10 on the snap builders, maybe the others?

Reply:
Ah, fair enough. Perhaps leave a note to clean up with Go 1.13 :)